xapian-letor/include/xapian-letor/ranker.h

```diff
@@ -82,7 +76,58 @@ class XAPIAN_VISIBILITY_DEFAULT Ranker {
     /// Compare function used to sort std::vector<Xapian::FeatureVector> by label values.
     static bool labelcomparer(const FeatureVector & firstfv, const FeatureVector& secondfv);
 
+  private:
+
+    /// Don't allow assignment.
+    void operator=(const Ranker &);
+
+    /// Don't allow copying.
+    Ranker(const Ranker & o);
+
+};
+
+/// ListNet Ranker class
+class XAPIAN_VISIBILITY_DEFAULT ListNETRanker: public Ranker {
+
+    /// Ranker parameters
+    std::vector<double> parameters;
+
+    /// Learning rate (Default is 0.0001)
+    double learning_rate;
+
+    /// Number of iterations (Default is 15)
+    int iterations;
+
+  public:
+
+    /// Default constructor
+    ListNETRanker();
+
+    /* Construct ListNet instance
+     * @param learn_rate       Learning rate
+     * @param num_interations  Number of iterations
+     */
+    ListNETRanker(double learn_rate, int num_interations);
+
+    /// Destructor
+    ~ListNETRanker();
+
+    /// Method to train the model.
+    void train_model();
+
+    /** Method to save model as an external file.
+     *  @param output_filename      Filename by which model is to be stored.
+     */
+    void save_model_to_file(const char* output_filename);
+
+    /// Method to load model as an external file.
+    void load_model_from_file(const char* model_filename);
+
+    /// Method to re-rank a std::vector<Xapian::FeatureVector> by using the model.
```

Comment:
While this comment is strictly true, it might be better as "Method to re-rank a list of FeatureVectors (each representing a Xapian::Document) by using the model" or similar? However more to think about for now, as I think this will need more detailed help (since it's one of the core methods for LTR).